### ISSUE TITLE
Bug: consistent-return does not into account exhaustiveness checks

### ISSUE THREAD

**Issue by SimonSchick** [typescript-eslint/typescript-eslint#8785]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.3&fileType=.tsx&code=MYewdgzgLgBFCm0YF4YAoBuAuGByAhrjAD54BGuAlDmSCADbz5goB8MA3gFAwwQDuASyjAAFugyVOPXjGD4I8PISwzZMAE7woAVw0soGnfADcauQqW4Kq9by279MAGb56is7wC%2BXL0A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra1oifLGhN0URNGgdokcGAC%2BIJUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const test = (v: 'a' | 'b'): boolean => {
  switch (v) {
    case 'a':
      return true;
    case 'b':
      return false;
  }
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/consistent-return": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Eslint really shouldn't flag these since adding additional code would be marked as unreachable.

### Actual Result

Eslint flags this code but there's little room to actually fix the error.

### Additional Info

This is especially annoying in large switch/case statements.

**Comment by bradzacher** [comment]
This is intentionally not handled.
Our extension rule exists purely to add minor QOL improvements.
Tracking exhaustiveness involves massive complexity and so it's not something we'll build. 
We're not looking to replicate typescript features here. `noImplicitReturns` exists and does everything you want already.

https://github.com/typescript-eslint/typescript-eslint/pull/8759 is open and will add a warning to the docs.
